Add contribution docs to the website (#2272)
AI-Hypercomputer/maxtext: docs/development.md

```{include} ../CONTRIBUTING.md
```

## Contributing documentation

### Building the documentation locally (optional)

If you are writing documentation for MaxText, you may want to preview the documentation site locally. First, make sure you install the necessary dependencies. You can do this by navigating to your local clone of the MaxText repo and running:
```bash
pip install -r requirements_docs.txt
```

Once the dependencies are installed, you can navigate to the `docs/` folder and run:

```bash
sphinx-build -b html . _build/html
```

> This will generate the documentation in the `docs/_build/html` directory.

Author: ```suggestion
This will generate the documentation in the `docs/_build/html` directory.

If you are adding a new document, make sure it is included in the `toctree` directive corresponding to the section where the new document should live. For example, if adding a new tutorial, make sure it is listed in [the `docs/tutorials.md`](https://github.com/AI-Hypercomputer/maxtext/blob/7070e8eecbea8951c8e5281219ce797c8df1441f/docs/tutorials.md?plain=1#L38).
```

```{include} ../CONTRIBUTING.md
```

## Contributing documentation

> ### Building the documentation locally
> 
> To build the documentation locally, make sure you install the necessary dependencies. You can do this by navigating to your local clone of the MaxText repo and running:
> 

Author: ```suggestion
### Building the documentation locally (optional)

If you are writing documentation for MaxText, you may want to preview the documentation site locally. First, make sure you install the necessary dependencies. You can do this by navigating to your local clone of the MaxText repo and running:
```

```bash
pip install -r requirements_docs.txt
```

Once the dependencies are installed, you can navigate to the `docs/` folder and run:

```bash
sphinx-build -b html . _build/html
```

> This will generate the documentation in the `docs/_build/html` directory.

Reviewer: Could you please clarify if this is needed for documentation deployment, or optional just for locally pre-viewing / testing the docs ?

Author: Let me know if this updated version is better.

Reviewer 2: Added a suggestion to make this a bit clearer.

Reviewer 2: Also, what about adding a note about how folks can check the built `html` either by opening these directly or using something like `python -m http.server -d docs/_build/html` ?

```{include} ../CONTRIBUTING.md
```

## Contributing documentation

The MaxText documentation website is built using [Sphinx](https://www.sphinx-doc.org) and [MyST](https://myst-parser.readthedocs.io/en/latest/). Documents are written in [MyST Markdown syntax](https://myst-parser.readthedocs.io/en/latest/syntax/typography.html#syntax-core).

### Building the documentation locally (optional)

> If you are writing documentation for MaxText, you may want to preview the documentation site locally. First, make sure you install the necessary dependencies. You can do this by navigating to your local clone of the MaxText repo and running:

Reviewer 2: ```suggestion
If you are writing documentation for MaxText, you may want to preview the documentation site locally to ensure things work as expected before a deployment to Read The Docs. 

First, make sure you install the necessary dependencies. You can do this by navigating to your local clone of the MaxText repo and running:
```

```bash
pip install -r requirements_docs.txt
```

Once the dependencies are installed, you can navigate to the `docs/` folder and run:

```bash
sphinx-build -b html . _build/html
```

This will generate the documentation in the `docs/_build/html` directory.

> If you are adding a new document, make sure it is included in the `toctree` directive corresponding to the section where the new document should live. For example, if adding a new tutorial, make sure it is listed in [the `docs/tutorials.md`](https://github.com/AI-Hypercomputer/maxtext/blob/7070e8eecbea8951c8e5281219ce797c8df1441f/docs/tutorials.md?plain=1#L38).

Reviewer 2: ```suggestion
### Adding new documentation files

If you are adding a new document, make sure it is included in the `toctree` directive corresponding to the section where the new document should live. For example, if adding a new tutorial, make sure it is listed in [the `docs/tutorials.md`](https://github.com/AI-Hypercomputer/maxtext/blob/7070e8eecbea8951c8e5281219ce797c8df1441f/docs/tutorials.md?plain=1#L38).
```

Reviewer 2: For consistency with the section above
